From c66a91ec878b75eea374dc3b6702cb5e5555b404 Mon Sep 17 00:00:00 2001
From: ye li <ye.li@nxp.com>
Date: Fri, 14 Oct 2016 10:29:14 +0800
Subject: [PATCH 1217/5242] MLK-13347 dts: mx6sll: enable DCP and hardware RNG
 for i.mx6sll

commit  4d2def432751f97d3136b807feee60d1d3049d46 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add DCP and RNG node in imx6sll.dtsi to enable them.

Signed-off-by: ye li <ye.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sll.dtsi |   11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6sll.dtsi b/arch/arm/boot/dts/imx6sll.dtsi
index f1826a9..095c9af 100644
--- a/arch/arm/boot/dts/imx6sll.dtsi
+++ b/arch/arm/boot/dts/imx6sll.dtsi
@@ -640,11 +640,13 @@
 			};
 
 			dcp: dcp@020fc000 {
+				compatible = "fsl,imx6sl-dcp";
 				reg = <0x020fc000 0x4000>;
 				interrupts = <GIC_SPI 99 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>;
-				status = "disabled";
+				clocks = <&clks IMX6SLL_CLK_DCP>;
+				clock-names = "dcp";
 			};
 		};
 
@@ -772,6 +774,13 @@
 				reg = <0x021b0000 0x4000>;
 			};
 
+			rngb: rngb@021b4000 {
+				compatible = "fsl,imx6sl-rng", "fsl,imx-rng", "imx-rng";
+				reg = <0x021b4000 0x4000>;
+				interrupts = <GIC_SPI 5 IRQ_TYPE_LEVEL_HIGH>;
+				clocks =  <&clks IMX6SLL_CLK_DUMMY>;
+			};
+
 			ocotp: ocotp-ctrl@021bc000 {
 				compatible = "fsl,imx6sll-ocotp", "syscon";
 				reg = <0x021bc000 0x4000>;
-- 
1.7.9.5

